js/web/components/Search.js

```diff
@@ -0,0 +1,319 @@
+    )
+  }, [])
+
+  useEffect(() => {
+    console.log('searchQuery', searchQuery)
+    console.log('searchResults', searchResults)
+
+    if (searchQuery) {
+      const query = searchQuery.q
+      setQuery(query)
+    }
+    if (searchResults) {
+      setResponse(searchResults)
+      setFetchedResponse(true)
+    }
+  }, [searchResults, searchQuery])
+
+  useEffect(() => {
+    const handleDropdown = (e) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(e.target) &&
+        !searchInputRef.current.contains(e.target)
+      ) {
+        setShowDropdown(false)
+      }
+    }
+
+    document.addEventListener('click', handleDropdown)
+
+    return () => {
+      document.removeEventListener('click', handleDropdown)
+    }
+  }, [showDropdown, dropdownRef])
+
+  useEffect(() => {
+    if (query && setShowDropdown) {
+      suggestions?.artists?.length > 0
+        ? (autoCompleteResults[0].visible = true)
+        : (autoCompleteResults[0].visible = false)
+      suggestions?.releases?.length > 0
+        ? (autoCompleteResults[1].visible = true)
+        : (autoCompleteResults[1].visible = false)
+      suggestions?.hubs?.length > 0
+        ? (autoCompleteResults[2].visible = true)
+        : (autoCompleteResults[2].visible = false)
+      setAutocompleteResults([...autoCompleteResults])
+    }
+  }, [suggestions])
+
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    setFetchedResponse(false)
+
+    if (
+      e.target.value !== null ||
+      e.target.value !== '' ||
+      e.target.value !== undefined
+    ) {
+      await NinaSdk.Search.withQuery(query).then(setResponse)
+      setFetchedResponse(true)
+      window.history.pushState(null, query, `?q=${query}`)
+    }
+    if (query === '') {
+      e.preventDefault()
+      e.stopPropagation()
+      return
+    }
+    setShowDropdown(false)
+  }
+
+  const autoCompleteUrl = 'https://dev.api.ninaprotocol.com/v1/suggestions'
+
+  const handleAutoComplete = async (query) => {
+    setLoading(true)
+    const response = await axios.post(autoCompleteUrl, { query })
+    if (query.length > 0) {
+      setSuggestions(response.data)
+    }
+    if (suggestions) {
+      setLoading(false)
+    }
+  }
+
+  const changeHandler = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    const search = e.target.value
+    console.log('searrch', search)
+    setQuery(search)
+    setShowDropdown(true)
+    if (query) {
+      handleAutoComplete(query)
+    }
+  }
+
+  const handleSuggestionsClick = async (search) => {
+    setFetchedResponse(false)
+    await NinaSdk.Search.withQuery(search).then(setResponse)
+    setFetchedResponse(true)
+    if (fetchedResponse) {
+      setShowDropdown(false)
+      window.history.pushState(null, search, `?q=${search}`)
+    }
+  }
+
+  const suggestionsHandler = (e) => {
+    e.preventDefault()
+    e.stopPropagation()
+    const clickedSuggestion = e.target.innerText
```

Comment:
can we can this something more descriptibe / tightly paired to `searchQuery` naming conventions?